pkg/apis/discovery/v1alpha1/clusterducktype_types.go

```diff
@@ -71,6 +72,10 @@ type ClusterDuckTypeSpec struct {
 	// Selectors is a list of selectors for CustomResourceDefinitions to identify a duck type.
 	// +optional
 	Selectors []CustomResourceDefinitionSelector `json:"selectors,omitempty"`
+
+	// ClusterRoleRef is a reference to the Aggregating ClusterRole used by the duck type to manage the ducks
+	// +optional
+	ClusterRoleRef rbacv1.RoleRef `json:"clusterRoleRef,omitempty"`
```

Comment:
I think `RoleRef` is a struct? in that case we need to make this a pointer so we can allow the `omitempty` setting let it be `nil`

Reply:
fixed